### DISCUSSION TITLE
Allow user to duplicate / clone habits

### DISCUSSION THREAD

**Discussion by firelite** [iSoron/uhabits#295] [Feature Requests]
Hi, 

Really loving the app so far! The first thing I felt was missing is the ability to duplicate an existing habit when creating a new habit.  Many of my habits had similar wording so I had to retype each one by memory.  It would be nice to have a quick way to copy existing habits and then modify as needed. 

Thanks!  Wonder if this is something others could use too.

**Comment by iSoron** [comment]
Thank you for the suggestion, @firelite. I agree that this would be an interesting feature.

**Comment by carsonskjerdal** [comment]
I will take a stride at adding this, shouldn't be too hard. Do we want the edit screen to come up automatically after duplicating?

**Comment by mr-flibble** [comment]
> I will take a stride at adding this, shouldn't be too hard. Do we want the edit screen to come up automatically after duplicating?

I think so, otherwise user will be forced to think which habit is original and which is a copy :)

**Comment by stheid** [comment]
just want to bump this quite inactive request up, as i think its still an importaint feature.

alternatively an import&export for single trackers would suffice.

**Reply by carsonskjerdal** [reply]
> I will take a stride at adding this, shouldn't be too hard. 

I was certainly out of my depth with this (at the time). I sadly no longer have the time to follow up on this feature.

**Comment by I-Dont-Remember** [comment]
A couple open questions for user experience:
- From the main list page, if you hold to select 1+ Habits - should you be able to duplicate multiple at once? In that case, would any of them open for editing? or should "Clone" be hidden if you have multiple selected?
- Should the clone be prefixed with an indicator, e.g. `(COPY) meditation` if `meditation` was the original title? I'm thinking prefix might be better than suffix, since on small screens the suffix can end up hidden on the right side and then the user still has to open the detail view to tell the clone from the original.

## Current PR branch behavior

Not locked in on this, but this is what [my branch](https://github.com/iSoron/uhabits/compare/dev...I-Dont-Remember:uhabits:feature/duplicate-habit?expand=1) currently does in case there are better approaches. Holding off on making PR until discussion is decided.

- Clone option available either from list view or from Habit detail screen
- Clone has a blank slate for history
- If only a single habit is cloned, then the Edit view of that modal is opened
- If multiple are cloned, no edit is opened.
- After cloning 1+ from list view, the list is refreshed with the clones _(see photo of `(Copy) <habit title>`)_


<p float="left">
<img width="200" height="450" alt="new Clone option in the menu" src="https://github.com/user-attachments/assets/dc1f931e-0ea2-49a2-a113-95eb8ce50db8" />
<img width="200" height="450" alt="Screenshot_20260406-213122" src="https://github.com/user-attachments/assets/71fb9afe-db93-4388-9c7a-3ba04c7e5f48" />
<img width="200" height="450" alt="Screenshot_20260406-215241" src="https://github.com/user-attachments/assets/50fb5384-0ddc-48c4-b283-60d321ef1fe7" />
</p>
